bin/generate_protein_and_entity_ids.py

```diff
@@ -95,22 +99,38 @@ def main(args=None):
 
                 proteins['protein_id']  = range(next_protein_id, next_protein_id + len(proteins))
                 next_protein_id += len(proteins)
-                proteins.rename(columns={'protein_tmp_id' : 'protein_orig_id'}, inplace=True)
 
-                entities = proteins[['entity_name']].drop_duplicates()
-                entities['entity_id'] = range(next_entity_id, next_entity_id + len(entities))
-                next_entity_id += len(entities)
-                entities['microbiome_id'] = mb_id
+                # Check if microbiome is coassembly
+                if len(microbiomes.groupby('microbiome_bare_id').get_group(mb_id)) != 1:
+                    all_entities = []
+                    # Iterate over coassemblies multiple weight tsv for one microbiome
+                    for coas_id in microbiomes.groupby('microbiome_bare_id').get_group(mb_id)['microbiome_id']:
+                        entities = pd.DataFrame()
+                        entities = proteins[['entity_name']].drop_duplicates()
+                        entities['entity_id'] = range(next_entity_id, next_entity_id + len(entities))
+                        next_entity_id += len(entities)
```

Comment:
this would lead to redundant `entity_id` entries across co-assemblies for the same entity, or? Wouldn't be the goal to use the same `entity_id` but have multiple entries with different `microbiome_ids`?

Reply:
For this I needed to also change the `finalize_microbiome_entities.py`, as it would create redundant microbiome_entity ids. But it's fixed now and one `entity_id` points to multiple `microbiome_id`s.

Reply:
(just redundant `entity_id` - `entity_name` entries, that can be dropped in )